### PR TITLE
test: add test for ipfs.pubsub.publish with a binary (non-utf8) buffer

### PR DESCRIPTION
**Pull request by vith**
This currently fails in js-ipfs-api.

issue: https://github.com/ipfs/js-ipfs-api/issues/569
fix PR: https://github.com/ipfs/js-ipfs-api/pull/570

### CODE REVIEW

**Comment by diasdavid** [comment]
Thank you for catching this and coming up with a solution, @vith! Really appreciate it :)

Mind fixing the linting for clean merge?

File: src/pubsub.js [hunk 1]
```diff
@@ -382,6 +382,46 @@ module.exports = (common) => {
...
+            const check = makeCheck(3, done)
+            const expectedHex = 'a36161636179656162830103056164a16466666666f4'
+            const buffer = Buffer.from(expectedHex, 'hex')
+
+            const sub1 = (msg) => {
+              try {
+                expect(msg.data.toString('hex')).to.be.eql(expectedHex)
+                expect(msg.from).to.be.eql(ipfs2.peerId.id)
+                check()
+              } catch (e) {
```
Comment (diasdavid) [inline src/pubsub.js]: use `err` so that standard can catch if missing

File: src/pubsub.js [hunk 2]
```diff
@@ -382,6 +382,46 @@ module.exports = (common) => {
...
+                ipfs1.pubsub.unsubscribe(topic, sub1)
+              }
+            }
+
+            const sub2 = (msg) => {
+              try {
+                expect(msg.data.toString('hex')).to.be.eql(expectedHex)
+                expect(msg.from).to.be.eql(ipfs2.peerId.id)
+                check()
+              } catch (e) {
```
Comment (diasdavid) [inline src/pubsub.js]: use `err` so that standard can catch if missing

File: src/pubsub.js [hunk 3]
```diff
@@ -382,6 +382,46 @@ module.exports = (common) => {
...
+                check(e)
+              } finally {
+                ipfs2.pubsub.unsubscribe(topic, sub2)
+              }
+            }
+
+            series([
+              (cb) => ipfs1.pubsub.subscribe(topic, sub1, cb),
+              (cb) => ipfs2.pubsub.subscribe(topic, sub2, cb),
+              (cb) => waitForPeers(ipfs2, topic, [ipfs1.peerId.id], cb),
```
Comment (diasdavid) [inline src/pubsub.js]: dangling comma

**Comment by vith** [comment]
Apologies, didn't realize my eslint editor integration wasn't working. Is that better?

**Comment by diasdavid** [comment]
perfect, thank you @vith!

**Pull request merged**